src/utils/work/seekWorkResponse.js

```diff
@@ -28,7 +28,8 @@ const getRemainingWorkerStartTime = (creationTimestamp) => {
 
 const seekFromS3 = async (taskName, signedUrl) => {
   const response = await fetch(signedUrl);
-  if (response.status === httpStatusCodes.NOT_FOUND) {
+  if (response.status === httpStatusCodes.NOT_FOUND
```

Comment:
Wouldn't that confuse us a bit. Can a message be added saying that its more like the file doesn't exist and not that its there but you don't have permissions?

Reply:
I understand that for work requests like `scType` and `runClustering`, which don't upload data to S3, getting `NOT_FOUND` or `FORBIDDEN` is ok, but for other types of work requests getting `NOT_FOUND` or `FORBIDDEN` is a genuine error and therefore is not ok. How does the code here distinguish between `scType` and `runClustering` -like work requests and others?

Reply:
Is it possible/realistic that for types of work requests like `scType` and `runClustering` in the cases when they now return `FORBIDDEN` or `NOT_FOUND`, we change the API to return something else to avoid situations like this when the UI has to treat seemingly invalid states as valid? Is it worth it?

Reply:
honestly, the whole work results thing is a mess, and now it's super hard to keep track of how results are sent back. Ideally, the worker wouldn't send the data back (through the signedURL) at all, so this issue wouldn't happen. 

However, this would either introduce "if" cases which don't look great on code (if sctype => don't return signedURL or anything) or a much deeper look at the worker to try to simplify things. This PR just fixes a regression introduced by the worker improvements (that 404/403 handling was explicitly removed) and I think should be merged as is but yeah I think this needs deeper discussion in the long-term.